scripts/create-api.py

```diff
@@ -1,36 +1,56 @@
 #!/usr/bin/env python
 
-import frontmatter
 import json
 import os
+import sys
 from collections import defaultdict
 
+import frontmatter
+import yaml
 
 ANNOTATIONS_DIR = os.path.join(os.path.dirname(__file__), "..", "annotations")
 
-data = defaultdict(lambda: defaultdict(lambda: {}))
+data = {}  # defaultdict(lambda: )
 
 apps = os.listdir(ANNOTATIONS_DIR)
 for app in apps:
+    app_dir = os.path.join(ANNOTATIONS_DIR, app)
+    valid_components = []
+    try:
+        metadata = yaml.load(open(os.path.join(app_dir, "metadata.yaml")))
+        data[app] = metadata
+        valid_components = metadata.get("components", []).keys()
+    except:
+        data[app] = {}
+        pass
+    data[app]["annotations"] = defaultdict(lambda: {})
     for annotation_type in ("metrics", "pings"):
-        annotation_dir = os.path.join(ANNOTATIONS_DIR, app, annotation_type)
+        annotation_dir = os.path.join(app_dir, annotation_type)
```

Comment:
Using `pathlib.Path` objects can clean a good bit of this up, avoiding the need for `os.path.join` calls in favor of `app_dir / annotation_type`, `os.listdir` calls can be replaced with `p.iterdir()`, etc.